src/formElementPlugins/checkbox/src/ApplicationView.tsx

```diff
@@ -23,13 +23,15 @@ const ApplicationView: React.FC<ApplicationViewProps> = ({
   onSave,
   Markdown,
   initialValue,
+  validationState
 }) => {
   const { getPluginStrings } = useLanguageProvider()
   const strings = getPluginStrings('checkbox')
   const { isEditable } = element
   const { label, description, checkboxes, type, layout, resetButton = false, keyMap } = parameters
 
   const [isFirstRender, setIsFirstRender] = useState(true)
+  const [toggled, setToggled] = useState(false)
```

Comment:
I can see where you're going with this -- making it so it doesn't immediately show up as invalid. Good idea, the only problem is this value gets reset every reload, so if you change page and go back to it, the warning has disappeared.

Also -- when you press the "Submit" button, the error message should appear then too, which is doesn't do with this approach.

Normally we check if the current response is `null` and use that to consider whether the user has "interacted" with the element yet or not. The problem with the checkboxes is that we save a response immediately, as "no selection" needs to be a valid selection for checkboxes.

I think what we might need to do for this is create a new parameter called "allowNoSelection", which defaults to "true". But when it's "false", then we *don't* save the response on initial load, so not selecting anything will be considered "invalid". 

This is a bit trickier than the initial issue and will require a bit of thorough checking for all the possible situations. Let's discuss.